js/views/message_list_view.js

```diff
@@ -17,31 +17,29 @@
             if (this.$el.scrollTop() === 0) {
                 this.$el.trigger('loadMore');
             }
-            if (this.bottomOffset === 0) {
+            if (this.atBottom()) {
                 this.$el.trigger('atBottom');
             } else if (this.bottomOffset > this.outerHeight) {
                 this.$el.trigger('farFromBottom');
             }
         },
+        atBottom: function() {
+            return this.bottomOffset < 20;
+        },
         measureScrollPosition: function() {
             if (this.el.scrollHeight === 0) { // hidden
                 return;
             }
             this.outerHeight = this.$el.outerHeight();
             this.scrollPosition = this.$el.scrollTop() + this.outerHeight;
             this.scrollHeight = this.el.scrollHeight;
-            this.shouldStickToBottom = this.scrollPosition === this.scrollHeight;
-            if (this.shouldStickToBottom) {
-                this.bottomOffset = 0;
-            } else {
-                this.bottomOffset = this.scrollHeight - this.$el.scrollTop() - this.outerHeight;
-            }
+            this.bottomOffset = this.scrollHeight - this.$el.scrollTop() - this.outerHeight;
```

Comment:
This chunk seems slightly tangential from the others. Are we sure we want to drop this if statement?

Git blame says it is related to scroll position jumping around when images are loading, so I guess just make sure you test with images, or large videos, and other tricky scroll corner cases, i.e., text field resizing, window resizing, short conversations that may or may not have scroll bars if you resize the window, scroll height changing due to loading older messages...